### PR TITLE
Fix #2197: Update constant naming to camelCase based on modern Swift conventions

### LINKED ISSUE #2197: Swift should use camelCase convention
This is a cleanup task for our project, no code logic should change.

### PR DESCRIPTION
**Pull request by freak4pc**
## Summary of Changes

This doesn't fix an existing issue, but I noticed that the naming conventions for many constants and properties are Swift 3/4 style (ClassCase), so I've went ahead an renamed them all to classCase based on modern Swift conventions.

It's a big PR, but the changes are quite minor in the broader sense.

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:
There should be nothing to test. The compiler and existing tests cover us well enough IMO.

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Comment by freak4pc** [comment]
By the way, might I suggest making sure Travis also works for PRs? you're missing a lot of testing otherwise :)

File: BraveRewardsUI/Create Wallet/CreateWalletViewController.swift [hunk 1]
```diff
@@ -75,8 +75,8 @@ class CreateWalletViewController: UIViewController {
   // MARK: - Actions
   
   func showFailureAlert() {
-    let alertController = UIAlertController(title: Strings.GenericErrorTitle, message: Strings.GenericErrorBody, preferredStyle: .alert)
-    alertController.addAction(UIAlertAction(title: Strings.OK, style: .default, handler: nil))
+    let alertController = UIAlertController(title: Strings.genericErrorTitle, message: Strings.genericErrorBody, preferredStyle: .alert)
+    alertController.addAction(UIAlertAction(title: Strings.oK, style: .default, handler: nil))
```
Comment (iccub) [inline BraveRewardsUI/Create Wallet/CreateWalletViewController.swift]: this must be either `ok` or `OK`, whatever sounds better

File: BraveRewardsUI/Localized Strings/Strings.swift [hunk 2]
```diff
@@ -7,290 +7,287 @@ import BraveShared
...
+  static let addFundsShowQRCode = NSLocalizedString("BraveRewardsAddFundsShowQRCode", bundle: .rewardsUI, value: "Show QR Code", comment: "")
+  static let rewardsOptInPrefix = NSLocalizedString("RewardsOptInPrefix", bundle: .rewardsUI, value: "Get ready to experience the next Internet.", comment: "")
+  static let disclaimerLearnMore = NSLocalizedString("BraveRewardsDisclaimerLearnMore", bundle: .rewardsUI, value: "Learn More", comment: "")
+  static let CLAIM = NSLocalizedString("CLAIM", bundle: .rewardsUI, value: "CLAIM", comment: "")
+  static let autoContributeNextDate = NSLocalizedString("BraveRewardsAutoContributeNextDate", bundle: .rewardsUI, value: "Next contribution date", comment: "")
+  static let monthlyTippingNextDate = NSLocalizedString("BraveRewardsMonthlyTippingNextDate", bundle: .rewardsUI, value: "Next monthly tip date", comment: "")
+  static let tippingMakeMonthly = NSLocalizedString("BraveRewardsTippingMakeMonthly", bundle: .rewardsUI, value: "Make this monthly", comment: "")
+  static let oK = NSLocalizedString("OK", bundle: .rewardsUI, value: "OK", comment: "")
```
Comment (iccub) [inline BraveRewardsUI/Localized Strings/Strings.swift]: see my previous comment

File: BraveShared/BraveStrings.swift [hunk 3]
```diff
@@ -17,156 +17,156 @@ import Shared
...
-    public static let SaveButtonTitle = NSLocalizedString("SaveButtonTitle", tableName: "BraveShared", bundle: Bundle.braveShared, value: "Save", comment: "Label for the button used to save data")
-    public static let Share = NSLocalizedString("CommonShare", tableName: "BraveShared", bundle: Bundle.braveShared, value: "Share", comment: "Text to select sharing something (example: image, video, URL)")
-    public static let Download = NSLocalizedString("CommonDownload", tableName: "BraveShared", bundle: Bundle.braveShared, value: "Download", comment: "Text to choose for downloading a file (example: saving an image to phone)")
-    public static let ShowLinkPreviewsActionTitle = NSLocalizedString("ShowLinkPreviewsActionTitle", tableName: "BraveShared", bundle: Bundle.braveShared, value: "Show Link Previews", comment: "Context menu item for showing link previews")
-    public static let HideLinkPreviewsActionTitle = NSLocalizedString("HideLinkPreviewsActionTitle", tableName: "BraveShared", bundle: Bundle.braveShared, value: "Hide Link Previews", comment: "Context menu item for hiding link previews")
+// MARK:-  Common Strings.here.
```
Comment (iccub) [inline BraveShared/BraveStrings.swift]: remove that extra dot in front of `here`

File: Client/Frontend/Settings/NewTabPageTableViewController.swift [hunk 4]
```diff
@@ -49,7 +49,7 @@ class NewTabPageTableViewController: TableViewController {
             rows.append(sponsoredRow)
         }
         
-        rows.append(BoolRow(title: Strings.NewTabPageSettingsAutoOpenKeyboard, option: Preferences.NewTabPage.autoOpenKeyboard))
+        rows.append(boolRow(title: Strings.newTabPageSettingsAutoOpenKeyboard, option: Preferences.NewTabPage.autoOpenKeyboard))
```
Comment (iccub) [inline Client/Frontend/Settings/NewTabPageTableViewController.swift]: the `boolRow` now looks like a class starting with small letter.
Could you rename it to `makeBoolRow` or something?

File: Client/WebAuthN/U2FExtensions.swift [hunk 5]
```diff
@@ -1463,8 +1463,8 @@ extension U2FExtensions: TabContentScript {
...
     public static let tryAgain = NSLocalizedString("tryAgain", tableName: "BraveShared", bundle: Bundle.braveShared, value: ", please try again.", comment: "Suffix for error strings")
-    public static let U2FRegistrationError = NSLocalizedString("U2FRegistrationError", tableName: "BraveShared", bundle: Bundle.braveShared, value: "Error registering your security key", comment: "Error handling U2F registration.") + tryAgain
-    public static let U2FAuthenticationError = NSLocalizedString("U2FAuthenticationError", tableName: "BraveShared", bundle: Bundle.braveShared, value: "Error authenticating your security key", comment: "Error handling U2F authentication.") + tryAgain
+    public static let u2FRegistrationError = NSLocalizedString("U2FRegistrationError", tableName: "BraveShared", bundle: Bundle.braveShared, value: "Error registering your security key", comment: "Error handling U2F registration.") + tryAgain
+    public static let u2FAuthenticationError = NSLocalizedString("U2FAuthenticationError", tableName: "BraveShared", bundle: Bundle.braveShared, value: "Error authenticating your security key", comment: "Error handling U2F authentication.") + tryAgain
```
Comment (iccub) [inline Client/WebAuthN/U2FExtensions.swift]: Rename both to `U2F..` 
it should be always capitalized, like here https://www.yubico.com/authentication-standards/fido-u2f/

**Review by iccub — CHANGES_REQUESTED** [review]
wow, that was a lot of changes to look at

I'm still in talks with my team on whether we want to rename all to camel case now and to enable CI for external contributors

Found few minor issues if you would like to take a look at them, also the branch needs a rebase.
Once all is good I will try to merge that PR as soon as possible, otherwise almost each new merged PRs will result in merge conflicts on this branch

thank you for taking time working on this

File: Client/Frontend/Settings/NewTabPageTableViewController.swift (hunk 4 above)
Comment (freak4pc) [reply Client/Frontend/Settings/NewTabPageTableViewController.swift]: What about `Row.boolRow()`? 

```swift
extension Row {
    static func boolRow(title: String, option: Preferences.Option<Bool>, onValueChange: SwitchAccessoryView.ValueChange? = nil) -> Row {
        return Row(
            text: title,
            accessory: .view(SwitchAccessoryView(initialValue: option.value, valueChange: onValueChange ?? { option.value = $0 })),
            cellClass: MultilineValue1Cell.self,
            uuid: option.key
        )
    }
}
```

Would also allow the following short syntax:
image

Comment (freak4pc) [reply Client/Frontend/Settings/NewTabPageTableViewController.swift]: Did this for now, happy for your opinion.

**Comment by freak4pc** [comment]
@iccub Rebased and fixed. I also did another quick pass to fix constants name that had underscores in them and aligned them to the rest of the naming conventions.

**Review by iccub — APPROVED** [review]
Looks great, can you rebase one last time?
I can block incoming PRs until you rebase, if that doesn't take long

**Comment by freak4pc** [comment]
I'll make it quick ;)

**Comment by freak4pc** [comment]
@iccub should be good to go

**Review by iccub — APPROVED** [review]
Perfect, thank you for your quick response.
I will run your branch for a bit to make sure strings and translations are working properly and then merge it

Awesome job

**Comment by freak4pc** [comment]
Sounds good. Good thing we have a compiler to make life easy ;) 
Cheers!

**Pull request merged**